### PR TITLE
test: add tests for readline highlight functions

### LINKED ISSUE #318: [Test Coverage] Refactor readline highlight processing for testability
## Summary
Extract pure functions from readline highlight processing to improve testability and increase coverage from 0% to ~12% for `cli_readline.go`.

## Parent Issue
Part of #248 - Improve test coverage for readline and system variables

## Current Status
- **File**: `cli_readline.go` (303 lines)
- **Current coverage**: Almost 0%
- **Target functions**:
  - `lexerHighlighterWithError` (line 153): 10.0% → 90%
  - `tokenHighlighter` (line 180): 50.0% → 100%
  - `kindHighlighter` (line 186): 50.0% → 100%
  - `commentHighlighter` (line 196): 25.0% → 100%

### 1. Extract Pure Functions
Create new file `cli_readline_highlight.go` with pure functions:
- Token classification logic
- Highlight rule matching
- Color mapping logic

### 2. Refactor Existing Functions
Modify existing functions to use the extracted pure functions while maintaining the same interface.

### 3. Add Comprehensive Tests
Create `cli_readline_highlight_test.go` with tests for:
- All token types (keywords, strings, numbers, params, idents)
- Error cases (unclosed strings, unclosed comments)
- Edge cases (empty input, special characters)

## Expected Coverage Gain
- **Lines to cover**: ~36 lines
- **Impact on cli_readline.go**: 0% → ~12%
- **New test file**: `cli_readline_highlight_test.go`

## Success Criteria
- [ ] Pure functions extracted without changing external behavior
- [ ] All highlight-related functions have >90% coverage
- [ ] Tests cover all token types and error cases
- [ ] No regression in existing functionality

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
This PR adds comprehensive tests for readline highlight functions to improve test coverage without modifying the existing implementation.

## Approach
- Added tests directly to `cli_readline_test.go` 
- No refactoring of existing code required
- Maintains the simplicity and cohesion of the original implementation

## Coverage Improvements
| Function | Before | After |
|----------|--------|-------|
| lexerHighlighterWithError | 10.0% | 100.0% |
| tokenHighlighter | 50.0% | 100.0% |
| kindHighlighter | 50.0% | 100.0% |
| commentHighlighter | 25.0% | 100.0% |
| **cli_readline.go overall** | ~0% | **21.5%** |

## Tests Added
- Tests for all token types (keywords, strings, numbers, parameters, identifiers)
- Tests for error cases (unclosed strings, unclosed comments)
- Tests for edge cases (empty input, no matches)
- Tests for color sequence generation

## Benefits
- Achieved target coverage (12%) and exceeded it to 21.5%
- No code changes required - purely additive
- Tests run in the same package, allowing access to unexported functions
- Simple and maintainable approach

Fixes #318
EOF < /dev/null

### CODE REVIEW

File: cli_readline_test.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+		input     string
+		predicate func(token.Token) bool
+		wantCount int
+	}{
+		{
+			name:  "highlight keywords",
+			input: "SELECT FROM WHERE",
+			predicate: func(tok token.Token) bool {
+				return alnumRe.MatchString(string(tok.Kind))
+			},
```
[the remark this answers is not in the document]
Comment (apstndb) [reply cli_readline_test.go:84]: Thank you for the review. However, this observation is incorrect. Both `cli_readline.go` and `cli_readline_test.go` are in `package main` (not a separate test package), so the test has full access to the unexported package-level variable `alnumRe`. 

This is a common Go testing pattern - tests in the same package can access unexported symbols. The test is working correctly and no changes are needed.
EOF < /dev/null

File: cli_readline_test.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+func TestDefaultHighlights(t *testing.T) {
+	// Save and restore color setting
+	originalNoColor := color.NoColor
+	color.NoColor = false
+	defer func() { color.NoColor = originalNoColor }()
+	
+	// Verify defaultHighlights is properly initialized
+	if len(defaultHighlights) == 0 {
+		t.Fatal("defaultHighlights is empty")
```
[the remark this answers is not in the document]
Comment (apstndb) [reply cli_readline_test.go:274]: Thank you for the suggestion. While I understand the concern about `t.Fatal` stopping further tests, in this specific case, using `t.Fatal` is the correct choice because:

1. If `defaultHighlights` is empty, it indicates a fundamental initialization failure in the highlight system
2. All subsequent assertions in this test function depend on `defaultHighlights` having content
3. Continuing would only produce cascading failures that don't provide additional useful information

The test is structured so that if this critical initialization fails, we want to stop immediately rather than produce misleading subsequent errors. This is a deliberate design choice for this particular test.
EOF < /dev/null
